e2e/cypress/integration/account_settings/sidebar/channel_switcher_spec.js

```diff
@@ -76,7 +76,7 @@ describe('Account Settings > Sidebar > Channel Switcher', () => {
         cy.get('#quickSwitchHint').should('be.visible').should('contain', 'Type to find a channel. Use UP/DOWN to browse, ENTER to select, ESC to dismiss.');
 
         // # Type CTRL/CMD+shift+L
-        cy.get('#quickSwitchInput').cmdOrCtrlShortcut('{shift}L');
+        cy.findByRole('textbox', {name: 'quick switch input'}).cmdOrCtrlShortcut('{shift}L');
```

Comment:
Could we not use a `data-testid` tag to find this instead of searching by an `aria-label`?

Reply:
We wanted to test it like on how the user would find the element (either visually or thru assistive technology) and less under the hood. So, it's better to have an aria-label (if possible/applicable). If not, then we can use other approach like `data-testid`.

Reply:
I think it would make sense to maybe have one test to make sure the user can find the element that way, but not as a general practice. I think for the other tests it would make more sense to use a `data-testid` attribute so we can reliably find the element. Since we're using an automated testing tool anyways I don't see this being too much of a sacrifice wrt. emulating the UX. 3/5

Reply:
If it's going to test other [attribute of the element](https://github.com/mattermost/mattermost-developer-documentation/pull/745/files#diff-b28dacdaa9e85745c4e44fbafc016a2f3a7b6077b31fedce8dab7b94d34847c6R35-R45) like `cy.findByTestId('testId`).should('have.attr', ...)` or accessing a section within, then I agree. But in most cases, we'd like to use queries that are "accessible to everyone". We're documenting this approach. Feel free to comment as well at https://github.com/mattermost/mattermost-developer-documentation/pull/745/files#diff-b28dacdaa9e85745c4e44fbafc016a2f3a7b6077b31fedce8dab7b94d34847c6R10. It's based from [Testing library's "which query to use"](https://testing-library.com/docs/guide-which-query/).

Reply:
Didn't know that was a thing, go with that then.